### PR TITLE
Add all the study progress stuff including a student chapter view.

### PR DESCRIPTION
**Pull request by DasGraMMa**
This includes study progress updates..:
* ..to the front page: course cards now include the (rudimentary) study progress for the user.
* ..to a chapter view: each content item now includes its study progress status (one of: No Progress, Started, Completed)
  In addition, clicking on a chapter (currently in both, lecturer and student view) will open the ScreenChapterStudent
* ..with a new screen: the ScreenChapterStudent, which currently includes a playlist of all videos and handles setting the state of the study progress tracker according to the video progress, completing it when a heartbeat is sent ~15 seconds before the video is ended.

Additionally, this pushed the ProgressService, a nice wrapper around the whole study progress tracker API and its objects for ease of use of the progress objects. Examples can be found in the code above.

### CODE REVIEW

File: src/components/ChapterComponent.tsx [hunk 1]
```diff
@@ -125,6 +134,132 @@ export const ChapterComponent: React.FC<ChapterComponentProps> = (props) => {
...
+
+                                            )/}
+                                    </View>
+                                </TouchableOpacity>
+                            );
+                        })}
+                </View>
+            </View>
+        </TouchableOpacity>
+    );*/
```
Comment (Noeliel) [inline src/components/ChapterComponent.tsx:191]: Kann man das hier löschen?

File: src/api/endpoints/EndpointsProgress.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+     * @param errorMsg An error message.
+     * @returns
+     */
+    public getContentProgress(
+        getRequest: RequestInit,
+        trackerId: string,
+        successMsg?: string,
+        errorMsg?: string
+    ): Promise<IContentProgressTracker> {
+        const urlToUse = `${this.url}/${ProgressUrlPart.MID_CONTENT}/${trackerId}`;
```
Comment (Noeliel) [inline src/api/endpoints/EndpointsProgress.ts:61]: Ist das \` \` für den String Absicht? Scheint mir ein bisschen ungewöhnlich, aber ich kenn mich mit TS nicht so mega gut aus.

File: src/api/endpoints/EndpointsProgress.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+     */
+    public updateContentProgress(
+        putRequest: RequestInit,
+        trackerId: string,
+        progress: number,
+        successMsg?: string,
+        errorMsg?: string
+    ): Promise<IContentProgressTracker> {
+        const urlToUse =
+            `${this.url}/${ProgressUrlPart.MID_CONTENT}/${trackerId}/${ProgressUrlPart.SUF_PROGRESS}` +
```
Comment (Noeliel) [inline src/api/endpoints/EndpointsProgress.ts:107]: Wie oben

File: src/components/ChapterComponent.tsx (hunk 1 above)
Comment (DasGraMMa) [reply src/components/ChapterComponent.tsx:191]: Oh stimmt, das hatte ich zum Übertragen auskommentiert, wollte ich noch entfernen, danke. :D

File: src/api/endpoints/EndpointsProgress.ts (hunk 2 above)
Comment (DasGraMMa) [reply src/api/endpoints/EndpointsProgress.ts:61]: Yep! Das ist zum Interpolieren von Strings (dem `${code}`-Teil darin), ein bisschen schöner als `"" + code + ""`, finde ich.

**Comment by DasGraMMa** [comment]
Update: removed the commented out code in ChapterComponent.

**Review by anna-riesch — COMMENTED** [review]
Looks fine

**Review by anna-riesch — APPROVED** [review]

**Pull request merged**